Docs: Expand GraphQL error codes (#7160)
apollographql/router: docs/source/routing/errors.mdx

</Note>

## GraphQL Error codes

GraphQL error codes can appear in client responses under `errors[].extensions.code`, which is an established convention found in [GraphQL error extensions](https://spec.graphql.org/October2021/#sec-Errors.Error-result-format). Learn how to see these error codes in Studio via [extended error metrics](/graphos/routing/graphos-reporting#enabling-extended-error-reporting).

<PropertyList kind="errCodes">

<Property name="CANNOT_SEND_PQ_ID_AND_BODY">

> The operation could not be executed because sending a persisted query ID and a

Reviewer: ```suggestion
The operation wan not executed because sending a persisted query ID and a
```

Author: Made this change w/o the typo.

The response from a subgraph did not match the GraphQL schema.

</Property>
<Property name="GATEWAY_TIMEOUT">

There request timed out when fetching data from a connector service.

</Property>
<Property name="GRAPHQL_VALIDATION_FAILED">

The operation failed during GraphQL validation.

</Property>
<Property name="GRAPHQL_UNKNOWN_OPERATION_NAME">

The operation could not be executed because the operation name was invalid or
did not match an operation in the query document.

</Property>
<Property name="HTTP_CLIENT_ERROR">

There was an error at the HTTP transport layer when fetching data from a
connector service.

</Property>
<Property name="MAX_ALIASES_LIMIT">

> The operation was not executed due to exceeding the max_aliases limit.

Reviewer: ```suggestion
The operation was not executed due to exceeding the `max_aliases` limit.
```

</Property>
<Property name="MAX_DEPTH_LIMIT">

> The operation was not executed due to exceeding the max_depth limit.

Reviewer: ```suggestion
The operation was not executed due to exceeding the `max_depth` limit.
```

</Property>
<Property name="MAX_HEIGHT_LIMIT">

The operation was not executed due to exceeding the max_height limit.

</Property>
<Property name="MAX_ROOT_FIELDS_LIMIT">

> The operation was not executed due to exceeding the max_root_fields limit.

Reviewer: ```suggestion
The operation was not executed due to exceeding the `max_root_fields` limit.
```

</Property>
<Property name="QUERY_NOT_IN_SAFELIST">

The operation was not executed because it was not found in the persisted query
safelist.

</Property>
<Property name="PARSING_ERROR">

The operation could not be parsed as GraphQL.

</Property>
<Property name="PERSISTED_QUERY_HASH_MISMATCH">

The operation was not executed due to a mismatch with the automatic persisted query (APQ) protocol.
There was an attempt to store this operation in the APQ cache, but the provided hash did not match the operation.

</Property>
<Property name="PERSISTED_QUERY_NOT_FOUND">

> The operation was not executed since it was not found in the automatic persisted query (APQ) cache.  

Reviewer: ```suggestion
The operation was not executed because it was not found in the automatic persisted query (APQ) cache.  
```
